src/app.rs

```diff
@@ -2312,6 +2339,66 @@ impl Application for App {
         let cosmic_theme::Spacing { space_xxs, .. } = theme::active().cosmic().spacing;
 
         let dialog = match dialog_page {
+            DialogPage::Compress { paths, to, name } => {
+                let mut dialog = widget::dialog(fl!("create-archive"));
+
+                let complete_maybe = if name.is_empty() {
+                    None
+                } else if name == "." || name == ".." {
+                    dialog = dialog.tertiary_action(widget::text::body(fl!(
+                        "name-invalid",
+                        filename = name.as_str()
+                    )));
+                    None
+                } else if name.contains('/') {
+                    dialog = dialog.tertiary_action(widget::text::body(fl!("name-no-slashes")));
+                    None
+                } else {
+                    let path = to.join(name);
```

Comment:
This does not test the existence properly, as the name does not have .zip added

Reply:
Good catch. I just added the extension here.